src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -130,6 +130,11 @@ public bool AllowInsecureTls
             set
             {
                 if (_isFrozen) { throw new InvalidOperationException("MongoClientSettings is frozen."); }
+                if (value && SslSettings != null)
+                {
+                    // Otherwise, the user will have to manually set CheckCertificateRevocation to false
+                    SslSettings.CheckCertificateRevocation = false;
+                }
```

Comment:
Probably better (as suggested in MongoClientSettings als):

    if (value)
    {
        _sslSettings = _sslSettings ?? new SslSettings();
        _sslSettings.CheckCertificateRevocation = false;
    }